pressbooks/pressbooks-book
# h5p listing page improvements

This PR solves https://github.com/pressbooks/pressbooks/issues/1996

We have a h5p-listing page on pressbooks, it's a page where we can see the total of h5p activities of a book the current implementation was printing all the activities in the first page load this is very exhaustive for server performance I added pagination to get chunks of 20 activities per page.

### How to test

1. Add a couple of h5p activities in your book (or clone https://integrations.pressbooks.network/h5p-clone, which has 30+ H5P activities in it)
2. Visit `https://pressbooks.test/yourbook/h5p-listing` the h5p listing page
3. You should see the total of activities under the page title
4. If you want to test the pagination you can change the default `per_page` parameter of the `inc/helpers/namespace.php` `get_h5p_activities` function

Notes:

I removed the transient caching because I think is no more needed with the pagination because the queries are going to be fast
